src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,83 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
 
 public class TournamentStats {
+    private static final int SORT_ASCENDING = 1;
+    private static final int SORT_DESCENDING = 2;
+    private static final int SORT_BY_FIRSTNAME = 1;
+    private static final int SORT_BY_LASTNAME = 2;
+    private static final int SORT_BY_RESULT = 3;
+    private List<Player> players = new LinkedList<>();
 
     void run(Scanner scanner) {
-        // tutaj dodaj swoje rozwiązanie
-        // użyj przekazanego scannera do wczytywania wartości
+        String userOption = null;
+        do {
+            System.out.println("Podaj wynik kolejnego gracza (lub STOP):");
+            userOption = scanner.nextLine();
+            players.addAll(getPlayers(userOption));
+        } while (!userOption.equals("STOP"));
+        Comparator<Player> comparator = sortByParameter(scanner);
 
+        Comparator<Player> comparator1 = sortAscendingOrDescending(scanner, comparator);
+
+        players.sort(comparator1);
+
+        try {
+            savePlayersToFile("stats.csv");
+        } catch (IOException e) {
+            System.err.println("Nie udalo sie zapisac danych do pliku.");
+        }
+    }
+
+    private static Comparator<Player> sortAscendingOrDescending(Scanner scanner, Comparator<Player> comparator) {
+        System.out.println("Sortować rosnąco czy malejąco? (1 - rosnąco, 2 - malejąco)");
+        int sortingOption = scanner.nextInt();
+        if (sortingOption == SORT_DESCENDING) {
+            return comparator = comparator.reversed();
+        } else if (sortingOption == SORT_ASCENDING) {
+            return comparator;
+        }
+        return comparator;
+    }
+
+    private void savePlayersToFile(String fileName) throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(fileName));
+        for (Player player : players) {
+            writer.write(player.toCsv());
+            writer.newLine();
+        }
+        writer.close();
```

Comment:
Co prawda zamykasz ten plik później, ale w przypadku błędu może pozostać niezamknięty. Najlepiej wprowadź try-with-resources https://youtu.be/LRwSHA9YDgY